### PR TITLE
Annotate and enrich outgoing request with SAP Passport headers

### PR DESCRIPTION
**Pull request by Poussinette**
**Description**

DO NOT MERGE THIS PR. This pull request is meant to build the application gateway image that is needed to test a local kyma PoC instance with SAP Passport handling.

Changes proposed in this pull request:

- ...
- ...
- ...

**Related issue(s)**

### CODE REVIEW

File: components/application-gateway/cmd/applicationgateway/options.go [hunk 1]
```diff
@@ -29,28 +32,33 @@ func parseArgs() *options {
 	proxyTimeout := flag.Int("proxyTimeout", 10, "Timeout for proxy call.")
 	requestLogging := flag.Bool("requestLogging", false, "Flag for logging incoming requests.")
 	proxyCacheTTL := flag.Int("proxyCacheTTL", 120, "TTL, in seconds, for proxy cache of Remote API information")
+	annotatePassportHeaders := flag.Bool("annotatePassportHeaders", false, "Flag to indicate if we should annotate passport headers to outgoing request")
```
Comment (akgalwas) [inline components/application-gateway/cmd/applicationgateway/options.go:35]: I'm not convinced that Application Gateway is the right place to add this functionality. Application Gateway is designed to be a generic component that is responsible for authenticating requests. We should strive to keep it simple and not litter the design with specific use cases.

I would consider the following solutions instead:

- Using Service Catalog Bindings and implementing header annotation in Serverless Function. Service Catalog offers Redis binding. You could implement a Server-less Function that will fetch  key value from database and add headers before calling Application Gateway. 

- Extending Application Gateway with generic mechanism for enriching requests.There is already #6346 issue that proposes such a mechanism.  Adding passport headers is a specific case that could be implemented with the mentioned mechanism.

File: components/application-gateway/internal/proxy/proxy.go [hunk 2]
```diff
@@ -57,6 +63,12 @@ func New(
 	csrfTokenStrategyFactory csrf.TokenStrategyFactory,
 	config Config,
 	configRepository proxyconfig.TargetConfigProvider) ProxyHandler {
+	var passportEnricher *passport.RequestEnricher
+	if config.AnnotatePassportHeaders {
```
Comment (akgalwas) [inline components/application-gateway/internal/proxy/proxy.go:67]: I would rather introduce some new abstraction here that will be responsible for enriching requests (similar to Authorization Strategy Factory and CSRF Token Strategy Factory).

**Comment by kwiatekus** [comment]
@Poussinette please check if you still need this PR

**Pull request closed without merging**